operator/pkg/model/ingestion/ingress.go

```diff
@@ -0,0 +1,199 @@
+		for _, path := range rule.HTTP.Paths {
+
+			route := model.HTTPRoute{}
+
+			switch *path.PathType {
+			case slim_networkingv1.PathTypeExact:
+				route.PathMatch.Exact = path.Path
+			case slim_networkingv1.PathTypePrefix:
+				route.PathMatch.Prefix = path.Path
+			case slim_networkingv1.PathTypeImplementationSpecific:
+				route.PathMatch.Regex = path.Path
+			}
+
+			backend := model.Backend{
+				Name:      path.Backend.Service.Name,
+				Namespace: ing.Namespace,
+			}
+			if path.Backend.Service != nil {
+				backend.Port = &model.BackendPort{}
+				if path.Backend.Service.Port.Name != "" {
+					backend.Port.Name = path.Backend.Service.Port.Name
+				}
+				if path.Backend.Service.Port.Number != 0 {
+					backend.Port.Port = uint32(path.Backend.Service.Port.Number)
+				}
+			}
+			route.Backends = append(route.Backends, backend)
+			l.Routes = append(l.Routes, route)
+		}
+
+		insecureListenerMap[host] = l
+	}
+
+	secureListenerMap := make(map[string]model.HTTPListener)
+
+	// First, we check for TLS config, and set them up with Listeners to return.
+	for _, tlsConfig := range ing.Spec.TLS {
+
+		for _, host := range tlsConfig.Hosts {
+
+			var l model.HTTPListener
+
+			l, ok := secureListenerMap[host]
+			if !ok {
+				l, ok = insecureListenerMap[host]
+				if !ok {
+					l, ok = insecureListenerMap["*"]
+					if !ok {
+						continue
+					}
+				}
+			}
+
+			l.Sources = model.AddSource(l.Sources, sourceResource)
+			if tlsConfig.SecretName != "" {
+				l.TLS = &model.TLSSecret{
+					Name: tlsConfig.SecretName,
+					// Secret has to be in the same namespace as the Ingress.
+					Namespace: ing.Namespace,
+				}
+
+			}
+			l.Port = 443
+			l.Hostname = host
+			secureListenerMap[host] = l
+
+			defaultListener, ok := insecureListenerMap["*"]
+			if ok {
+				// A default listener already exists, each Host in TLSConfig.Hosts
+				// needs to have a Listener configured that's a copy of it.
+				if tlsConfig.SecretName != "" {
+					defaultListener.TLS = &model.TLSSecret{
+						Name: tlsConfig.SecretName,
+						// Secret has to be in the same namespace as the Ingress.
+						Namespace: ing.Namespace,
+					}
+				}
+				defaultListener.Hostname = host
+				defaultListener.Port = 443
+				secureListenerMap[host] = defaultListener
+
+			}
+		}
+
+	}
+
+	listenerSlice := make([]model.HTTPListener, 0, len(insecureListenerMap)+len(secureListenerMap))
+	listenerSlice = appendValuesInKeyOrder(insecureListenerMap, listenerSlice)
+	listenerSlice = appendValuesInKeyOrder(secureListenerMap, listenerSlice)
+
+	return listenerSlice
+
+}
+
+// appendValuesInKeyOrder ensures that the slice of listeners is stably sorted by
+// appending the values of the map in order of the keys to the appendSlice.
+func appendValuesInKeyOrder(listenerMap map[string]model.HTTPListener, appendSlice []model.HTTPListener) []model.HTTPListener {
+
+	var keys []string
+
+	for key := range listenerMap {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
```

Comment:
AFAIK `sort.Strings` is not guaranteed to be stable. But if I got it right, I think that here even an unstable sort should be ok. I'd just remove _stably_ from the comment to avoid possible confusion.

Reply:
> sort.Strings is not guaranteed to be stable
can you elaborate a little more on the above? I think as long as the result is sorted, we should be ok :thinking:

Reply:
Yep, I agree that we need to simply sort the result. I was a little bit confused by the word _stably_ in the comment. At first, it made me thought that a [stable sort](https://en.wikipedia.org/wiki/Category:Stable_sorts) (e.g. [sort.SliceStable](https://pkg.go.dev/sort#SliceStable)) was needed. That's why I was suggesting to remove "stably" from the comment, but maybe it's just me.